content/en/docs/develop/porting.md

```diff
@@ -614,23 +614,333 @@ To make a patch:
+1. To register patches against Unikraft's build system such that they are applied before the compilation of all source files, simply indicate it in the library's `Makefile.uk`:
 
    ```Makefile
    # Add or edit ~/workspace/libs/iperf3/Makefile.uk
    LIBIPERF3_PATCHDIR = $(LIBIPERF3_BASE)/patches
    ```
 
 This concludes the necessary steps to port an application to Unikraft "from first principles".
+
+### Relevant Files to the Build System
+
+In order for an application or library to work with Unikraft, you need to provide some specific files:
+
+1. **Makefile**:
+   Used to specify where the main Unikraft repository is with respect to the application's repository, as well as repositories for any external libraries the application needs to use.
+   The `Makefile` is not required when porting an external library.
+
+1. **Makefile.uk**:
+   Used to specify which sources to build (and optionally where to fetch them from), include paths, flags and any application-specific targets.
+
+1. **Config.uk**:
+   A [Kconfig-like](https://www.kernel.org/doc/html/latest/kbuild/kconfig-language.html) snippet used to populate Unikraft's menu with application-specific options.
+
+1. **exportsyms.uk**:
+   A text file where each line contains the name of one symbol that should be exported to other libraries.
+   This file usually contains only `main` for an application that is developed/ported as a single library to Unikraft.
+   If an `exportsyms.uk` file is not present, all symbols will be exported.
+
+1. **extra.ld**:
+   [Optional] Contains an amendment to the main linker script.
+
+In the tutorial below, we will focus on files belonging to an application (i.e. not to a library).
+The process of porting a library is very similar, just change the `APP` prefix of the variable names to `LIB`.
+
+### Makefile
+
+The Makefile is generally short and simple and might remind you of Linux kernel modules that are built off-tree.
+For most applications the Makefile should contain no more than the following:
+
+```Makefile
+UK_ROOT  ?= $(PWD)/../../unikraft
+UK_LIBS  ?= $(PWD)/../../libs
+UK_PLATS ?= $(PWD)/../../plats
+LIBS  := $(UK_LIBS)/lib1:$(UK_LIBS)/lib2:$(UK_LIBS)/libN
+PLATS ?=
+
+all:
+        @make -C $(UK_ROOT) A=$(PWD) L=$(LIBS) P=$(PLATS)
+
+$(MAKECMDGOALS):
+        @make -C $(UK_ROOT) A=$(PWD) L=$(LIBS) P=$(PLATS) $(MAKECMDGOALS)
+```
+
+You can notice that the `Makefile` is just a wrapper around the [`Unikraft core Makefile`](https://github.com/unikraft/unikraft/blob/staging/Makefile).
+You can get the same result if you use:
+
+```console
+$ make -C ../../unikraft A=. L=../../libs/lib1:../../libs/lib2:../../libs/libN
+```
+
+### Makefile.uk
+
+The Unikraft build system provides a number of functions and macros to make it easier to write the `Makefile.uk` file.
+Also, please ensure that all variables that you define begin with `APP[NAME]_` or `LIB[NAME]_` (e.g., `APPHELLOWORLD_`) so that they are properly namespaced.
```

Comment:
```suggestion
Also, ensure that all variables that you define begin with `APP[NAME]_` or `LIB[NAME]_` (e.g., `APPHELLOWORLD_`) so that they are properly namespaced.
```